inc/utility/class-handlestransfers.php

```diff
@@ -0,0 +1,348 @@
+				<?php wp_nonce_field( 'bulk-tags' ); ?>
+				<div class="form-field <?php echo self::TAXONOMY ?>-prefix-wrap">
+					<input type="file" name="import_file" />
+					<?php echo $messages['hint'] ?? ''; ?>
+				</div>
+				<p class="submit">
+					<button type="submit" class="button button-primary">
+						<?php echo __( 'Import', 'pressbooks' ); ?>
+					</button>
+				</p>
+			</form>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Imports a CSV file content and creates the taxonomy terms.
+	 *
+	 * @return void
+	 */
+	public function importCsv() {
+		// If no file was imported.
+		$upload = $this->handleUpload();
+
+		if ( ! $upload ) {
+			return;
+		}
+
+		$items = $this->getImportableItems( $upload );
+
+		unlink( $upload['file'] );
+
+		if ( ! $items ) {
+			return;
+		}
+
+		$this->handleImport( $items );
+
+		$_SESSION['pb_notices'][] = __( 'Successfully imported.', 'pressbooks' );
+	}
+
+	/**
+	 * Generates and download a CSV file with all the selected resources.
+	 *
+	 * @param $ids
+	 * @return void
+	 */
+	public function exportCsv( $ids ) {
+		$items = $this->getExportableItems( $ids );
+
+		if ( empty( $items ) ) {
+			return;
+		}
+
+		$content = $this->generateCsvContent( $items );
+
+		$this->downloadCsv( $content );
+	}
+
+	/**
+	 * Return a list of taxonomy terms that should be exported.
+	 *
+	 * @param $ids
+	 * @return array
+	 */
+	public function getExportableItems( $ids ) {
+		$items = [];
+		$fields = $this->getTransferableFields();
+
+		foreach ( $ids as $id ) {
+			$term = get_term( $id, self::TAXONOMY );
+			$term_meta = get_term_meta( $term->term_id );
+
+			$item = [
+				'name' => $term->name,
+				'slug' => $term->slug,
+			];
+
+			foreach ( $fields as $field ) {
+				$value = $term_meta[ $field ] ?? [];
+
+				$item[ $field ] = $value[0] ?? '';
+			}
+
+			$items[] = $item;
+		}
+
+		return $items;
+	}
+
+	/**
+	 * Generate the CSV content for the given array of items.
+	 *
+	 * @param array $items
+	 * @return false|string
+	 */
+	public function generateCsvContent( $items ) {
+		ob_start();
+
+		$df = fopen( 'php://output', 'w' );
+
+		fputcsv( $df, array_keys( $items[0] ) );
+
+		foreach ( $items as $row ) {
+			fputcsv( $df, $row );
+		}
+
+		fclose( $df );
+
+		return ob_get_clean();
+	}
+
+	/**
+	 * Download the content as a CSV file.
+	 *
+	 * @param string $content
+	 * @return void
+	 */
+	public function downloadCsv( $content ) {
+		$filename = self::TAXONOMY . '-list-' . time() . '.csv';
+
+		$file = create_tmp_file();
+
+		put_contents( $file, $content );
+
+		force_download( $file, false, $filename );
+
+		if ( ! defined( 'WP_TESTS_MULTISITE' ) ) {
+			exit;
+		}
+	}
+
+	/**
+	 * @return array|false
+	 */
+	public function handleUpload() {
+		if ( empty( $_FILES['import_file']['name'] ) ) {
+			return false;
+		}
+
+		if ( $_FILES['import_file']['type'] !== 'text/csv' ) {
+			$_SESSION['pb_errors'][] = __( 'Sorry, this file type is not permitted for security reasons.' );
```

Comment:
Please, confirm this message with others with Steel, just in case some English syntax issue. 
Also, I think for translation reasons, it should contains `'pressbooks'` as a second parameter, example:
```
__( 'Sorry, this file type is not permitted for security reasons.' , 'pressbooks');
```

Reply:
This string is perfect. I think it's already included in the WordPress project: https://github.com/WordPress/wordpress-develop/blob/5f5895bf61dfe62890e296ad7acd81aad9bde88b/src/wp-admin/includes/file.php#L934. Not sure if we need to take special action to 'inherit' localized strings from WP or if it happens already ... See https://developer.wordpress.org/themes/functionality/internationalization/#loading-translations for ideas?

Reply:
I think it happens already. I used the same approach as in here https://github.com/pressbooks/pressbooks/blob/dev/inc/modules/import/class-import.php#L421-L424